Error while checking build/O2/fullCI for fc8a0643d806c5645f724a0fdba4898c19ce2660 at 2024-08-27 05:08:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
/sw/SOURCES/O2/13356-slc8_x86-64/0/Detectors/ITSMFT/ITS/reconstruction/src/ClustererTask.cxx:36:16: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
/sw/SOURCES/O2/13356-slc8_x86-64/0/Detectors/ITSMFT/MFT/simulation/src/DigitizerTask.cxx:36:16: error: use '= default' to define a trivial destructor [modernize-use-equals-default]
/sw/SOURCES/O2/13356-slc8_x86-64/0/Detectors/ITSMFT/ITS/workflow/src/ThresholdAggregatorSpec.cxx:27:25: error: use '= default' to define a trivial default constructor [modernize-use-equals-default]
++ [[ 0 == 0 ]]
++ exit 1
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13356/fc8a0643d806c5645f724a0fdba4898c19ce2660/build_O2_fullCI/pretty.html).

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -1,4 +1,3 @@
-
 // Copyright 2019-2020 CERN and copyright holders of ALICE O2.
 // See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
 // All rights not expressly granted are reserved.

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -1,3 +1,4 @@
+
 // Copyright 2019-2020 CERN and copyright holders of ALICE O2.
 // See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
 // All rights not expressly granted are reserved.
@@ -202,23 +203,15 @@ class NoiseMap
     return std::ceil((1. + 1. / t) / (relErr * relErr));
   }
 
-  NoiseMap merge(NoiseMap* prev)
+  NoiseMap merge(const NoiseMap* prev)
   {
     int incre = 0;
     for (size_t i = 0; i < (int)mNoisyPixels.size(); ++i) {
       for (const auto& prev_np : prev->mNoisyPixels[i]) { // only enters this for loop if the "i" chip exists.
-        bool existsInCurrentMap = false;
-        for (const auto& current_np : mNoisyPixels[i]) {
-          if (prev_np.first == current_np.first) {
-            existsInCurrentMap = true;
-            break;
-          }
-        } // end of for loop on elements of previous noise map
-
-        if (!existsInCurrentMap) {
-          incre++;
-          mNoisyPixels[i][prev_np.first] = prev_np.second;
-        }
+	if (mNoisyPixels[i].find(prev_np.first) == mNoisyPixels[i].end()) {
+	  mNoisyPixels[i][prev_np.first] = prev_np.second;
+	  incre++;
+	}
       } // end of for loop on elements of previous noise map
     }   // end of for loop on i (chip ID)
     return (mNoisyPixels);

--- a/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
+++ b/DataFormats/Detectors/ITSMFT/common/include/DataFormatsITSMFT/NoiseMap.h
@@ -207,12 +207,12 @@ class NoiseMap
     int incre = 0;
     for (size_t i = 0; i < (int)mNoisyPixels.size(); ++i) {
       for (const auto& prev_np : prev->mNoisyPixels[i]) { // only enters this for loop if the "i" chip exists.
-	if (mNoisyPixels[i].find(prev_np.first) == mNoisyPixels[i].end()) {
-	  mNoisyPixels[i][prev_np.first] = prev_np.second;
-	  incre++;
-	}
+        if (mNoisyPixels[i].find(prev_np.first) == mNoisyPixels[i].end()) {
+          mNoisyPixels[i][prev_np.first] = prev_np.second;
+          incre++;
+        }
       } // end of for loop on elements of previous noise map
-    }   // end of for loop on i (chip ID)
+    } // end of for loop on i (chip ID)
     return (mNoisyPixels);
   } // end of void merge
 

--- a/Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx
+++ b/Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx
@@ -299,11 +299,14 @@ void NoiseCalibratorSpec::sendOutputCcdbMerge(DataAllocator& output)
   map<std::string, std::string> filter;
   auto* payloadPrev1 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, -1, &headers);
   long validtime = std::stol(headers["Valid-From"]);
-  validtime = validtime - 60;
-  auto* payloadPrev2 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, validtime, &headers);
-  auto bufferPL = payloadPrev2->merge(payloadPrev1);
-  auto mergedPL = payload.merge(&bufferPL);
-  // payload.merge(payloadPrev);
+  auto mergedPL = payload;
+  if(validtime>0)
+    {
+      validtime = validtime - 60;
+      auto* payloadPrev2 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, validtime, &headers);
+      auto bufferPL = payloadPrev2->merge(payloadPrev1);
+      mergedPL = payload.merge(&bufferPL);
+    }
   o2::ccdb::CcdbObjectInfo info(mPathMerge, "NoiseMap", "noise.root", meta, tstart, tend);
   auto flName = o2::ccdb::CcdbApi::generateFileName("noise");
   auto image = o2::ccdb::CcdbApi::createObjectImage(&mergedPL, &info);

--- a/Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx
+++ b/Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorSpec.cxx
@@ -300,13 +300,12 @@ void NoiseCalibratorSpec::sendOutputCcdbMerge(DataAllocator& output)
   auto* payloadPrev1 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, -1, &headers);
   long validtime = std::stol(headers["Valid-From"]);
   auto mergedPL = payload;
-  if(validtime>0)
-    {
-      validtime = validtime - 60;
-      auto* payloadPrev2 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, validtime, &headers);
-      auto bufferPL = payloadPrev2->merge(payloadPrev1);
-      mergedPL = payload.merge(&bufferPL);
-    }
+  if (validtime > 0) {
+    validtime = validtime - 60;
+    auto* payloadPrev2 = api.retrieveFromTFileAny<o2::itsmft::NoiseMap>(mPath, filter, validtime, &headers);
+    auto bufferPL = payloadPrev2->merge(payloadPrev1);
+    mergedPL = payload.merge(&bufferPL);
+  }
   o2::ccdb::CcdbObjectInfo info(mPathMerge, "NoiseMap", "noise.root", meta, tstart, tend);
   auto flName = o2::ccdb::CcdbApi::generateFileName("noise");
   auto image = o2::ccdb::CcdbApi::createObjectImage(&mergedPL, &info);

